scripts/src/pullrequest/metadata.py

```diff
@@ -0,0 +1,150 @@
+import re
+import sys
+import json
+
+from . import prartifact
+from tools import gitutils
+from reporegex import matchers
+from argparse import ArgumentParser
+
+
+class NoMatchesError(Exception):
+    def __init__(self):
+        super().__init__(
+            "no file modifications within the chart submission path were detected in this pr"
+        )
+
+
+class MultipleMatchesError(Exception):
```

Comment:
We already have checks in place (e.g. [here](https://github.com/komish/charts-development/blob/3f3fa8485b6d5e2a11641b89a6d1e757620f732f/scripts/src/checkprcontent/checkpr.py#L159) ) to ensure that a PR relates to only one chart.

Is this needed here ? Could we find a way to put the code in common ? This kind of relates to #296

Reply:
This library definitely has common code, but focuses just on the extraction of category, org, chart, etc. This is just one component of the linked code ( checkprcontent ), which also checks tarball naming, if relevant.

I do think this library can be leveraged there, but the additional checks need to be accounted for in some way, and I'm not ready to introduce changes to that code path if it's not strictly necessary. At least not as a part of this larger bugfix effort.

I'd vote for pursuing this in #296 as a future action item.